src/System.Runtime.Extensions/tests/System/Environment.SetEnvironmentVariable.cs

```diff
@@ -20,72 +28,149 @@ public void NullVariableThrowsArgumentNull()
+        [InlineData(EnvironmentVariableTarget.Machine)]
+        [InlineData(EnvironmentVariableTarget.User)]
+        public void Default(EnvironmentVariableTarget target)
+        {
+            const string varName = "Test_SetEnvironmentVariable_Default";
+            const string value = "true";
+
+            ExecuteAgainstTarget(target,
+            () =>
+            {
+                Environment.SetEnvironmentVariable(varName, value, target);
+                Assert.Equal(IsSupportedTarget(target) ? value : null,
+                    Environment.GetEnvironmentVariable(varName, target));
+            },
+            () =>
+            {
+                // Clear the test variable
+                Environment.SetEnvironmentVariable(varName, null, target);
+            });
+        }
+
+
+        [Theory]
+        [InlineData(EnvironmentVariableTarget.Process)]
+        [InlineData(EnvironmentVariableTarget.Machine)]
+        [InlineData(EnvironmentVariableTarget.User)]
+        public void ModifyEnvironmentVariable(EnvironmentVariableTarget target)
         {
             const string varName = "Test_ModifyEnvironmentVariable";
             const string value = "false";
 
-            try
+            ExecuteAgainstTarget(target,
+            () =>
             {
                 // First set the value to something and then change it and ensure that it gets modified.
-                Environment.SetEnvironmentVariable(varName, "true");
-                Environment.SetEnvironmentVariable(varName, value);
+                Environment.SetEnvironmentVariable(varName, "true", target);
+
+                Environment.SetEnvironmentVariable(varName, value, target);
 
                 // Check whether the variable exists.
-                Assert.Equal(value, Environment.GetEnvironmentVariable(varName));
-            }
-            finally
+                Assert.Equal(IsSupportedTarget(target) ? value : null, Environment.GetEnvironmentVariable(varName, target));
+            },
+            () =>
             {
                 // Clear the test variable
                 Environment.SetEnvironmentVariable(varName, null);
-            }
+            });
         }
 
-        [Fact]
-        public void DeleteEnvironmentVariable()
+        [Theory]
+        [InlineData(EnvironmentVariableTarget.Process)]
+        [InlineData(EnvironmentVariableTarget.Machine)]
+        [InlineData(EnvironmentVariableTarget.User)]
+        public void ModifyEnvironmentVariable_AndEnumerate(EnvironmentVariableTarget target)
+        {
+            const string varName = "Test_ModifyEnvironmentVariable_AndEnumerate";
+            const string value = "false";
+
+            ExecuteAgainstTarget(target,
+            () =>
+            {
+                // First set the value to something and then change it and ensure that it gets modified.
+                Environment.SetEnvironmentVariable(varName, "true", target);
```

Comment:
Yeah- unfortunately the matrix here is a little complicated. Breaking out into several tests for each of these was another option- but not one I wanted to go with both from a volume perspective and how awkward the conditioning would be as our current model doesn't have a clean way to do this.